internal/ethapi/api.go

```diff
@@ -642,14 +642,11 @@ func (s *PublicBlockChainAPI) doCall(ctx context.Context, args CallArgs, blockNr
 	// Setup context so it may be cancelled the call has completed
 	// or, in case of unmetered gas, setup a context with a timeout.
 	var cancel context.CancelFunc
-	if vmCfg.DisableGasMetering {
-		ctx, cancel = context.WithTimeout(ctx, time.Second*5)
-	} else {
-		ctx, cancel = context.WithCancel(ctx)
-	}
+	ctx, cancel = context.WithTimeout(ctx, time.Second*5)
```

Comment:
I hope you're aware that doCall is used for gas estimation, too. I'm unsure if the timeout and increased default gas can be an issue there.

Reply:
I hate these hard coded crap.

Gas estimation is based on binary search. I.e. it always needs an upper limit, which is set to either what the user supplies, or the current block limit. Both cases should be fine from a runtime perspective. But yes, this does tag on an extra timeout, which might be undersirable (heck, why do even do a non configurable hard coded timeout in the first place).